src/OnlineRetailPortal.Core/Models/Product.cs

```diff
@@ -38,7 +38,7 @@ public Product(IProductStore productStore)
             this.productStore = productStore;
         }
 
-        public async Task<GetProductsServiceResponse> GetProductsAsync(GetProductsServiceRequest serviceRequest)
+        public static async Task<GetProductsCoreResponce> GetProductsAsync(GetProductsServiceRequest serviceRequest)
```

Comment:
This should be getproductsresponse 
You need not to add a word core with it as this class is not tied up with core models
Also why the method is static